tests/test-nullspace.C

```diff
@@ -56,38 +95,40 @@ bool check(Field& F, FFLAS::FFLAS_SIDE side, size_t m, size_t n, size_t r, typen
     typename Field::Element_ptr NS;
     FFPACK::NullSpaceBasis(F, side, m, n, ACopy, lda, NS, ldns, NSdim);
 
-    if (DEBUG_LOG) {
-        std::cout << std::endl;
-        std::cout << "A: " << m << "x" << n << " (rank " << r << ")" << std::endl;
-        std::cout << "NS: " << NSdim << std::endl;
-    }
+#if defined(__FFLAS_FFPACK_DEBUG)
+    std::cout << std::endl;
+    std::cout << "A: " << m << "x" << n << " (rank " << r << ")" << std::endl;
+    std::cout << "NS: " << NSdim << std::endl;
+#endif
 
     if (side == FFLAS::FFLAS_SIDE::FflasRight) {
         // Right nullspace dimension + Rank == Matrix column dimension
         if (NSdim + r != n) return false;
 
+        // Ensure nullspace is full rank
+        auto NSCopy = FFLAS::fflas_new(F, n, NSdim);
+        FFLAS::fassign(F, n, NSdim, NS, NSdim, NSCopy, NSdim);
```

Comment:
FYI: there is an `FFPACK::IsSingular` method!

Reply:
`FFPACK::IsSingular` doesn't check for full rank of non-square. See #175